Return last ticket with query-spice

We use this to implement spice seemless migration. The current spice code
reconnect to the migrated VM using the original ticket. So we need 
a way to read the original ticket.

Limits: This only works for a single spice session.


Index: new/ui/spice-core.c
===================================================================
--- new.orig/ui/spice-core.c	2013-11-27 10:51:33.000000000 +0100
+++ new/ui/spice-core.c	2013-11-27 11:43:08.000000000 +0100
@@ -539,6 +539,11 @@
     micro = SPICE_SERVER_VERSION & 0xff;
     info->compiled_version = g_strdup_printf("%d.%d.%d", major, minor, micro);
 
+    if (auth_passwd) {
+        info->has_ticket = true;
+        info->ticket =  g_strdup(auth_passwd);
+    }
+
     if (port) {
         info->has_port = true;
         info->port = port;
Index: new/qapi-schema.json
===================================================================
--- new.orig/qapi-schema.json	2013-11-27 11:41:45.000000000 +0100
+++ new/qapi-schema.json	2013-11-27 11:43:08.000000000 +0100
@@ -1209,11 +1209,14 @@
 #
 # @channels: a list of @SpiceChannel for each active spice channel
 #
+# @ticket: #optional The last ticket set with set_password
+#
 # Since: 0.14.0
 ##
 { 'type': 'SpiceInfo',
   'data': {'enabled': 'bool', 'migrated': 'bool', '*host': 'str', '*port': 'int',
            '*tls-port': 'int', '*auth': 'str', '*compiled-version': 'str',
+           '*ticket': 'str',
            'mouse-mode': 'SpiceQueryMouseMode', '*channels': ['SpiceChannel']} }
 
 ##
